packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -604,8 +645,13 @@ class BootOrder extends Component {
   }
 }
 
+@connectStore(() => ({ hosts: createGetObjectsOfType('host') }), {
+  withRef: true,
+})
 class MigrateVdiModalBody extends Component {
   static propTypes = {
+    hostSrShouldBeOn: PropTypes.string,
+    isSrIllegal: PropTypes.func.isRequired,
```

Comment:
You also pass `hostSrShouldBeOn`. mutualise the selector instead of pass it.

Reply:
I need it to get host's name and display it in an error message.

Reply:
i refered to `isSrIllegal`.

Reply:
And maybe found a better name for it like `isSrIncompatible`.